src/protagonist/Portal/Features/Batches/Requests/ParseCsv.cs

```diff
@@ -0,0 +1,198 @@
+                        MediaType = "image/jp2"
+                    });
+                }
+                catch (BadDataException badDataEx)
+                {
+                    return ParseCsvResult.Failure(
+                        $"CSV read error: bad data found in file at line {badDataEx.Context.Parser.Row}, row {badDataEx.Context.Reader.CurrentIndex}");
+                }
+                catch (MissingFieldException missingFieldEx)
+                {
+                    return ParseCsvResult.Failure(
+                        $"CSV read error: line {missingFieldEx.Context.Parser.Row} in file is missing fields");
+                }
+                catch (TypeConverterException typeEx)
+                {
+                    var fieldIndex = typeEx.Context.Reader.CurrentIndex - 1;
+                    var currentLine = csv.GetField(1);
+                    readErrors.Add($"Line {currentLine}: could not parse {ImageIngestModel.FieldNames[fieldIndex]} value ({typeEx.Text})");
+                }
+                catch (Exception ex)
+                {
+                    logger.LogError(ex, "Error parsing CSV file");
+                    return ParseCsvResult.Failure("An error occured while parsing the CSV file");
+                }
+            }
+        }
+        
+        // Return an error if parsing fails
+        if (readErrors.Any()) 
+        {
+            return ParseCsvResult.Failure(readErrors.ToArray());
+        }
+        
+        // Order images
+        var images = distinctRows
+            .OrderBy(image => image.Key)
+            .Select(image => image.Value)
+            .ToList();
+        
+        // Split the read images into multiple batches if they surpass the maximum batch size
+        var batches = images.Chunk(maxBatchSize).ToList();
+        for (var i = 0; i < batches.Count; i++)
+        {
+            var collection = new HydraCollection<Image>()
+            {
+                Members = batches[i].ToArray()
+            };
+            try
+            {
+                await dlcsClient.CreateBatch(collection);
+            }
+            catch (DlcsException dlcsEx) // Forward errors from the API
+            {
+                // Get the range of images where the exception may have occured:
+                var imagesStart = (maxBatchSize * i) + 1;
+                var imagesEnd = (maxBatchSize * i) + batches[i].Length;
+                return ParseCsvResult.Failure($"DLCS Error in images {imagesStart}-{imagesEnd}: {dlcsEx.Message}");
+            }
+            catch(Exception ex)
+            {
+                logger.LogError(ex, "Error posting batch to DLCS");
+                return ParseCsvResult.Failure($"DLCS Error: An error occurred while posting this batch to the DLCS");
+            }
+        }
+        
+        return ParseCsvResult.Success();
+    }
+}
+
+public class ImageIngestModel
+{
+    public static readonly string[] FieldNames =
```

Comment:
These `FieldNames` don't line up with `ImageIngestModel` - should they? There are 14 fields but 15 properties on model

Reply:
Good catch! It's missing `InitialOrigin`, will fix asap